examples/aws/main.tf

```diff
@@ -32,42 +31,39 @@ resource "gardener_shoot" "test_cluster" {
   }
 
   spec {
-    cloud {
-      profile = "aws"
-      region  = "eu-central-1"
-
-      secret_binding_ref {
-        name = "<secret_binding>"
-      }
+    cloud_profile_name  = "aws"
+    region              = "eu-central-1"
+    secret_binding_name = "<secret_binding>"
 
-      aws {
-        networks {
-          vpc {
-            cidr = "10.250.0.0/16"
-          }
+    networking {
+      type     = "calico"
+      nodes    = "10.250.0.0/19"
+      pods     = "100.96.0.0/11"
+      services = "100.64.0.0/13"
+    }
 
-          internal = ["10.250.112.0/22"]
-          public   = ["10.250.96.0/22"]
-          workers  = ["10.250.0.0/19"]
+    provider {
+      type = "aws"
+      worker {
+        name            = "cpu-worker"
+        max_surge       = 1
+        max_unavailable = 0
+        maximum         = 2
+        minimum         = 2
+        volume {
+          size = "50Gi"
+          type = "Standard_LRS"

```

Comment:
for spec.provier.worker, following values are required (machine and zones)

```           
 machine {
              type          = lookup(worker.value, "machine_type", "t3.medium")
              image {
                name = var.machine_image_name
                version = var.machine_image_version
              }
            volume {
                type     = lookup(worker.value, "volume_type", "gp2")
                size     = lookup(worker.value, "volume_size", "30Gi")
            }
}
            zones = lookup(worker.value, "zones",["us-east-1a", "us-east-1b", "us-east-1c",])
}
```

Reply:
This might be part of the old API because I'm getting those error when adding the volume bloc 
![image](https://user-images.githubusercontent.com/11049349/74247833-b43de180-4cb4-11ea-9dbf-9b76adbbd1b9.png)

Reply:
this is part of new api - see gardener api https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/apis/aws/v1alpha1/types_worker.go#L25

in terraform-provider-gardener code
machine - https://github.com/kyma-incubator/terraform-provider-gardener/blob/950d87f824530370f7e4fc751f716d981d33552d/shoot/schema_shoot.go#L68

volume - https://github.com/kyma-incubator/terraform-provider-gardener/blob/950d87f824530370f7e4fc751f716d981d33552d/shoot/schema_shoot.go#L144



you are getting this error as the `terraform-provider-gardener` you are using might not be the latest ,  recently there are lots of new PRs merged so please try rebase / merge with latest master branch to see if works...